src/pip/_internal/resolution/resolvelib/reporter.py

```diff
@@ -0,0 +1,45 @@
+from collections import defaultdict
+from logging import getLogger
+
+from pip._vendor.resolvelib.reporters import BaseReporter
+
+from pip._internal.utils.typing import MYPY_CHECK_RUNNING
+
+if MYPY_CHECK_RUNNING:
+    from typing import DefaultDict
+
+    from .base import Candidate
+
+
+logger = getLogger(__name__)
+
+
+class PipReporter(BaseReporter):
+
+    def __init__(self):
+        # type: () -> None
+        self.backtracks_by_package = defaultdict(int)  # type: DefaultDict[str, int]
+
+        self._messages_at_backtrack = {
+            8: (
+                "pip is looking at multiple versions of this package to determine "
+                "which version is compatible with other requirements. "
+                "This could take a while."
+            ),
```

Comment:
We (accidentally?) did a one-person user-test of this with @ewdurbin, who suggested we present a message at the first backtrack. I'm thinking of changing this to present the same message at 1 and shipping this.

I think everyone agreed that would a good change, so... let's do that!

/cc @ei8fdb @ewdurbin to raise concerns if there are any.

Reply:
I'm updating #8975 to include it.

Reply:
I've updated #8975 with the new messages, and printing <message 1> before the *first* backtrack. @pradyunsg please include the `INFO:` label at the beginning of the line.

Reply:
> > > Happy to add it in -- I wasn't sure what exact link to use and the proposals all had placeholder links.
> > > Lemme know and I'll add it in. :)
> > 
> > 
> > OK, Ill create one and let you know.
> 
> I've created a GH issue template PR for users to report backtracking. What are you're thoughts on using Github's URL shortener to shorten the issue template URL?

@pradyunsg I'm waiting for this [PR for backtracking GH issue template](https://github.com/pypa/pip/pull/9029) to be merged. 

When it is I can then provide you with a URL.